src/envoy/tcp/alpn_proxy/header/alpn_proxy_header.proto

```diff
@@ -0,0 +1,30 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package envoy.tcp.alpnproxy.header;
+
+option java_outer_classname = "AlpnProxyHeaderProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.tcp.alpnproxy.header";
+option go_package = "AlpnProxy";
+
+import "google/protobuf/struct.proto";
+
+// [#protodoc-title: AlpnProxyHeaderProto is sent on wire as initial bytes]
+message AlpnProxyHeaderProto {
```

Comment:
Is this proto needed? Can we use `google.protobuf.Struct` instead of it, or even better, make that proto customizable?

Reply:
I feel its good to have it so that later if we want to add a specific field say host name, it can be easily added?
what did you have in mind for making it customizable? using Any?

Reply:
Yes, using Any. We could use a type url in the header that defaults to struct.

Reply:
Used Any now.. please check the logic..

This is what am doing=> if Any contains a struct, I take that struct out and pass that as metadata struct value. If not then I construct a new metadata struct value whose key is type_url of any and value is any. 
Kinda torn between using any's value as compared to any itself in metadata struct value's value.. what do you think?